city_scrapers/spiders/il_liquor_control.py

```diff
@@ -0,0 +1,161 @@
+                end=self._parse_end(dt_str),
+                all_day=False,
+                time_notes='',
+                location=self._parse_location(response),
+                links=self._parse_link(
+                        response, self._parse_start(dt_str), minute_href),
+                source=response.url,
+                )
+
+        meeting["status"] = self._get_status(meeting)
+        meeting["id"] = self._get_id(meeting)
+
+        yield meeting
+
+    def _prev_meeting_wo_page(self, response, dt_object, minute_href):
+        meeting = Meeting(
+                title='Board Meeting',
+                description='',
+                classification=BOARD,
+                start=dt_object + datetime.timedelta(hours=13),
+                end=dt_object + datetime.timedelta(hours=16),
+                all_day=False,
+                time_notes='Meeting time is estimated.',
+                location={'address': '100 West Randolph 9-040 Chicago, IL', 'name': ''},
+                links=self._parse_link(
+                        response, dt_object, minute_href),
+                source=minute_href,
+                )
+
+        meeting["status"] = self._get_status(meeting)
+        meeting["id"] = self._get_id(meeting)
+
+        yield meeting
+
+    def _parse_title(self, response):
+        """Parse or generate meeting title."""
+        return response.css('p.soi-eventType::text').extract_first().strip()
+
+    def _parse_start(self, dt_str):
+        """Parse start datetime as a naive datetime object."""
+
+        return datetime.datetime.strptime(' '.join(dt_str.split()[:6]), '%A, %B %d, %Y %I:%M %p')
+
+    def _parse_end(self, dt_str):
+        """Parse end datetime as a naive datetime object. Added by pipeline if None"""
+        return datetime.datetime.strptime(
+            ' '.join(dt_str.split()[:4] + dt_str.split()[7:]), '%A, %B %d, %Y %I:%M %p')
+
+    def _parse_location(self, response):
+        """Parse or generate location."""
+        loc_str = ' '.join([
+                x.strip() for x in response.css('div.soi-eventlocation div::text').extract()
+            ])
+        return {"address": '{} Chicago, IL'.format(' '.join(loc_str.split()[:4])),
+                "name": "", }
+
+    def _parse_link(self, response, dt_object, minute_href):
+        """Use agenda_href if it is found from detail page. If not, try href using formatting
+        datetime. """
+        links = []
+        agenda_href = response.css("div.item.link-item.bullet a::attr(href)").extract()
+        hidden_agenda_href = ("https://www2.illinois.gov/ilcc/Events/EventDocuments/{0.month}"
+                              "-{0.day}-{0.year}%20ILCC%20Agenda.pdf".format(dt_object))
+        if agenda_href:
+            links.append({
+                    'title': 'Agenda',
+                    'href': agenda_href[0]
+                    })
+        elif requests.get(hidden_agenda_href).status_code == 200:
```

Comment:
I think it's fine to remove this and only get the agenda if it's available on the detail page. The detail page is more useful when we only have minutes